### PR TITLE
Replaced getStringArray and putStringArray with getStringList and putStringList in preferences

### PR DESCRIPTION
**Pull request by oscargus**
Also fixed an NPE in the XMP metadata preference tab.

### CODE REVIEW

File: src/main/java/net/sf/jabref/exporter/CustomExportList.java [hunk 1]
```diff
@@ -69,12 +71,12 @@ private void readPrefs() {
...
         int i = 0;
-        String[] s;
-        while ((s = Globals.prefs.getStringArray(JabRefPreferences.CUSTOM_EXPORT_FORMAT + i)) != null) {
+        List<String> s;
+        while (!((s = Globals.prefs.getStringList(JabRefPreferences.CUSTOM_EXPORT_FORMAT + i)).isEmpty())) {
             Optional<ExportFormat> format = createFormat(s);
             if (format.isPresent()) {
                 formats.put(format.get().getConsoleName(), format.get());
-                list.add(s);
+                list.add(s.toArray(new String[s.size()]));
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/exporter/CustomExportList.java]: you can use addAll

**Comment by simonharrer** [comment]
:+1: less nulls :)

File: src/main/java/net/sf/jabref/exporter/CustomExportList.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/exporter/CustomExportList.java]: I do not think so. I do not want to add the elements but the array. `EventList<String[]>`. Or am I missing something?

Comment (simonharrer) [inline src/main/java/net/sf/jabref/exporter/CustomExportList.java]: Ah, sorry. You are right. But why are arrays stored in a list? But this is probably due for another PR. ;)

**Pull request merged**